ingress-nginx-controller-1.12.yaml

```diff
@@ -571,6 +571,60 @@ subpackages:
           cp -p otel_ngx_module.so ${{targets.contextdir}}/etc/nginx/modules/
       - uses: strip
 
+  - name: ingress-nginx-custom-error-pages-${{vars.nginx-ingress-major-minor}}
+    description: OTel plugin for ingress nginx controller ${{vars.nginx-ingress-major-minor}}
+    dependencies:
+      provides:
+        - ingress-nginx-custom-error-pages=${{package.full-version}}
+    pipeline:
+      - working-directory: images/custom-error-pages/rootfs/
+        pipeline:
+          - runs: |
+              mkdir -p ${{targets.contextdir}}/usr/bin
+              mkdir -p /go/src/k8s.io/ingress-nginx/images/custom-error-pages
+              cp -r * /go/src/k8s.io/ingress-nginx/images/custom-error-pages/
+              cd /go/src/k8s.io/ingress-nginx/images/custom-error-pages
+              go get . && \
+              go build -o ${{targets.contextdir}}/usr/bin/nginx-errors .
+    test:
+      environment:
+        contents:
+          packages:
+            - wait-for-it
+            - curl
+      pipeline:
+        - runs: |
+            /usr/bin/nginx-errors &
+            wait-for-it -t 10 localhost:8080
+
+            # check healthz endpoint make sure it returns 200
+            http_code=$(curl -s -o /dev/null -w "%{http_code}" http://localhost:8080/healthz)
+
+            if [ "$http_code" != "200" ]; then
```

Comment:
Is the point of this to provide a custom error page? Should the test check for a 4xx or 5xx code and return a custom response, or am I misunderstanding the intent of this package?

Reply:
I added a simple 404 test.